test/services/composite-keys-manager.test.js

```diff
@@ -0,0 +1,76 @@
+import ComppositeKeyManager from '../../src/services/composite-keys-manager';
+
+describe('services > composite-keys-manager', () => {
+  describe('getPrimaryKeyValues', () => {
+    const compositeKeyManager = new ComppositeKeyManager();
+    it('should return one value for non composite key', () => {
+      expect.assertions(1);
+      const primaryKeyValues = compositeKeyManager.getPrimaryKeyValues('1');
+      expect(primaryKeyValues).toStrictEqual(['1']);
+    });
+    it('should return two values for composite key string with two values', () => {
+      expect.assertions(1);
+      const primaryKeyValues = compositeKeyManager.getPrimaryKeyValues('1-2');
+      expect(primaryKeyValues).toStrictEqual(['1', '2']);
+    });
+    it('should return null if `null` string is present', () => {
+      expect.assertions(1);
+      const primaryKeyValues = compositeKeyManager.getPrimaryKeyValues('1-null');
+      expect(primaryKeyValues).toStrictEqual(['1', null]);
+    });
+  });
+
+  describe('getRecordConditions', () => {
+    it('should return a where condition with one key for non composite key', () => {
+      expect.assertions(1);
+      const model = { primaryKeys: { id: {} } };
+      const compositeKeyManager = new ComppositeKeyManager(model);
+      const conditions = compositeKeyManager.getRecordConditions('1');
+      expect(conditions).toStrictEqual({ id: '1' });
+    });
+    it('should return a where condition with two keys for composite key', () => {
+      expect.assertions(1);
+      const model = { primaryKeys: { actorId: {}, filmId: {} } };
+      const compositeKeyManager = new ComppositeKeyManager(model);
+      const conditions = compositeKeyManager.getRecordConditions('1-2');
+      expect(conditions).toStrictEqual({ actorId: '1', filmId: '2' });
+    });
+  });
+
+  describe('getRecordsConditions', () => {
+    it('should return a where condition with one key for non composite key', () => {
+      expect.assertions(1);
+      const model = { primaryKeys: { actorId: {} } };
+      const compositeKeyManager = new ComppositeKeyManager(model);
+      const conditions = compositeKeyManager.getRecordsConditions(['1', '2']);
+      expect(conditions).toStrictEqual({ actorId: ['1', '2'] });
+    });
+    it('should return a where condition with two keys for composite key', () => {
+      expect.assertions(1);
+      const model = { primaryKeys: { actorId: {}, filmId: {} } };
+      const compositeKeyManager = new ComppositeKeyManager(model);
+      const conditions = compositeKeyManager.getRecordsConditions(['1-2', '3-4']);
+      expect(conditions).toStrictEqual({ actorId: ['1', '3'], filmId: ['2', '4'] });
```

Comment:
That looks hazardous at first sight.
Does it mean that records with ids `1-2` and `3-4` will generate conditions where we could find records with ids `1-4` and `3-2`?